qcodes/tests/dataset/test_doNd.py

```diff
@@ -148,6 +157,36 @@ def test_do0d_output_data(_param):
     assert loaded_data == np.array([_param.get()])
 
 
+@pytest.mark.usefixtures("temp_exp", "temp_db")
+@pytest.mark.parametrize("multiparamtype", [MultiSetPointParam,
+                                            Multi2DSetPointParam,
+                                            Multi2DSetPointParam2Sizes])
+@given(n_points_pws=hst.integers(min_value=1, max_value=1000))
+@settings(deadline=None)
+def test_do0d_verify_shape(_param, _param_complex, multiparamtype,
+                           dummyinstrument, n_points_pws):
+    arrayparam = ArraySetPointParam(name='arrayparam')
+    multiparam = multiparamtype(name='multiparam')
+    paramwsetpoints = dummyinstrument.A.dummy_parameter_with_setpoints
+    dummyinstrument.A.dummy_start(0)
+    dummyinstrument.A.dummy_stop(1)
+    dummyinstrument.A.dummy_n_points(n_points_pws)
+
+
+
+    results = do0d(arrayparam, multiparam, paramwsetpoints,
+                   _param, _param_complex,
+                   do_plot=False)
+    expected_shapes = {}
+    for i, name in enumerate(multiparam.full_names):
+        expected_shapes[name] = tuple(multiparam.shapes[i])
+    expected_shapes['arrayparam'] = tuple(arrayparam.shape)
+    expected_shapes['simple_parameter'] = ()
+    expected_shapes['simple_complex_parameter'] = ()
+    expected_shapes[paramwsetpoints.full_name] = (n_points_pws, )
+    assert results[0]._shapes == expected_shapes
```

Comment:
to use public interfaces in tests?
```suggestion
    assert results[0].description.shapes == expected_shapes
```